src/Tasks/ResolveProjectBase.cs

```diff
@@ -277,6 +277,7 @@ protected void AddSyntheticProjectReferences(string currentProjectAbsolutePath)
                     item.SetMetadata("LinkLibraryDependencies", "false");
                     item.SetMetadata("CopyLocal", "false");
                     item.SetMetadata("SkipGetTargetFrameworkProperties", "true");
+                    item.SetMetadata("GlobalPropertiesToRemove", "TargetFramework");
```

Comment:
Should we do RID as well?

Reply:
I thought about this as well but:
* building a solution with `-r RID` is prone to lots of issues anyway. But if for some reason it works, the synthetic reference wouldn't cause an additional RID-less build.
* bullding a project with `-r RID` doesn't have this synthetic reference
* a project with a fixed RID doesn't set it as a global property

It would only help if you'd have an additional p2p reference with `AdditionalProperties="RuntimeIdentifier=osx-x64"`.

But on the other hand, it shouldn't do any harm under normal circumstances.

Reply:
I find that convincing! Let's leave as is.